wrangler-core/src/main/java/io/cdap/directives/column/SetType.java

```diff
@@ -134,33 +134,35 @@ public Schema getOutputSchema(SchemaResolutionContext context) {
               if (field.getName().equals(col)) {
                 Integer outputScale = scale;
                 Integer outputPrecision = precision;
-                Schema fieldSchema = field.getSchema().getNonNullable();
-                Pair<Integer, Integer> scaleAndPrecision = getPrecisionAndScale(fieldSchema);
-                Integer inputSchemaScale = scaleAndPrecision.getSecond();
-                Integer inputSchemaPrecision = scaleAndPrecision.getFirst();
+                if (type.equalsIgnoreCase("decimal") && field.getSchema().isNullable()) {
```

Comment:
What happens when schema is non nullable but set type directive is applied with 'decimal' as the type?

Reply:
If the schema is non nullable it will skip this if condition and will go to the return statement in line no:167. By default outputScale and outputPrecision are the scale and precision specified by the user. In this case we don't validate if the user specified scale and precision values clash with the original scale and precision values of that column.